### PR TITLE
Add tap for Identity

### LINKED ISSUE #1942: tap for Identity
### Current Behavior

Identity only has chainFirst.

### Desired Behavior

Identity should have tap as chainFirst.

### Suggested Solution

Add tap to Identity.

### Who does this impact? Who is this for?

chainFirst is legacy and alias of tap.
Every Monad should have tap as chainFirst.

### Describe alternatives you've considered

None

### Additional context

None

### Your environment

| Software   | Version(s) |
| ---------- | ---------- |
| fp-ts      |   2.16.6   |
| TypeScript |   5.5.2    |

### PR DESCRIPTION
**Pull request by forno**
Close #1942

### CODE REVIEW

File: src/Identity.ts [hunk 1]
```diff
@@ -275,7 +275,18 @@ export const Monad: Monad1<URI> = {
  * Composes computations in sequence, using the return value of one computation to determine the next computation and
  * keeping only the result of the first.
  *
- * @category sequencing
+ * @category combinators
+ * @since 2.17.0
```
Comment (gcanti) [inline src/Identity.ts:279]:
```suggestion
 * @since 2.16.7
```

File: src/Identity.ts [hunk 2]
```diff
@@ -275,7 +275,18 @@ export const Monad: Monad1<URI> = {
...
+  <A, _>(f: (a: A) => Identity<_>): (self: Identity<A>) => Identity<A>
+} = /*#__PURE__*/ dual(2, tap_(Chain))
+
+/**
+ * Alias of `tap`
+ *
+ * @category legacy
  * @since 2.0.0
  */
 export const chainFirst: <A, B>(f: (a: A) => B) => (first: A) => A = /*#__PURE__*/ chainFirst_(Chain)
```
Comment (gcanti) [inline src/Identity.ts:292]:
```suggestion
export const chainFirst: <A, B>(f: (a: A) => B) => (first: A) => A = tap
```

**Review by gcanti — CHANGES_REQUESTED** [review]

File: src/Identity.ts [hunk 3]
```diff
@@ -275,10 +275,21 @@ export const Monad: Monad1<URI> = {
  * Composes computations in sequence, using the return value of one computation to determine the next computation and
  * keeping only the result of the first.
  *
- * @category sequencing
+ * @category combinators
+ * @since 2.16.0
```
Comment (gcanti) [inline src/Identity.ts:279]:
```suggestion
 * @since 2.16.7
```

**Review by gcanti — CHANGES_REQUESTED** [review]

File: src/Identity.ts (hunk 3 above)
Comment (forno) [reply src/Identity.ts:279]: Oh. sorry. I will fix soon.

**Review by gcanti — APPROVED** [review]

**Pull request merged**